client/driver/docker.go

```diff
@@ -1495,3 +1538,13 @@ func authOptionFrom(file, repo string) (*docker.AuthConfiguration, error) {
 
 	return apiAuthConfig, nil
 }
+
```

Comment:
Hey it would be great if we could actually reuse the auth config file. That file supports setting credential helpers per repo and we parse it right above: https://github.com/hashicorp/nomad/pull/2651/files#diff-a6757d41ad7cd7f37f0f0243e3575307R1524.

I would suggest we use that to determine if we should use a credential helper.